spek-ide-plugin/intellij-common/src/main/kotlin/org/spekframework/intellij/SpekRunConfigurationProducer.kt

```diff
@@ -31,14 +35,44 @@ abstract class SpekRunConfigurationProducer(val producerType: ProducerType, conf
 ) {
     override fun isConfigurationFromContext(configuration: SpekBaseRunConfiguration<*>,
                                             context: ConfigurationContext): Boolean {
-        val path = context.psiLocation?.let { extractPath(it, true) }
-        return configuration.path == path
+        val descriptor = context.psiLocation?.let {
+            val elementContext = PsiTreeUtil.getContextOfType(it, KtClassOrObject::class.java, KtCallExpression::class.java)
+            if (elementContext != null) {
+                if (elementContext is KtClassOrObject) {
+                    ScopeDescriptorCache.toDescriptor(elementContext)
+                } else if (elementContext is KtCallExpression) {
+                    ScopeDescriptorCache.toDescriptor(elementContext)
+                } else {
+                    null
+                }
+            } else {
+                null
+            }
+        }
+
+        if (descriptor == null) {
+            return false
+        }
+        return configuration.path == descriptor.path
```

Comment:
just `return configuration.path == descriptor?.path`? That way we can get rid of the if statement above